src/main/java/picard/illumina/IlluminaBasecallsToFastq.java

```diff
@@ -196,26 +199,40 @@
     public enum ReadNameFormat {
         CASAVA_1_8, ILLUMINA
     }
-    private final Map<String, AsyncClusterWriter> sampleBarcodeClusterWriterMap = new HashMap<>(1, 0.5f);
+
+    private final Map<String, Writer<ClusterData>> sampleBarcodeClusterWriterMap = new HashMap<String, Writer<ClusterData>>(1, 0.5f);
     final BclQualityEvaluationStrategy bclQualityEvaluationStrategy = new BclQualityEvaluationStrategy(MINIMUM_QUALITY);
     private ReadStructure readStructure;
     private BasecallsConverter<?> basecallsConverter;
     private static final Log log = Log.getInstance(IlluminaBasecallsToFastq.class);
     private final FastqWriterFactory fastqWriterFactory = new FastqWriterFactory();
     private ReadNameEncoder readNameEncoder;
+    private AsyncWriterPool writerPool;
 
     @Override
     protected int doWork() {
         initialize();
         final Set<String> barcodes = sampleBarcodeClusterWriterMap.keySet();
-        basecallsConverter.processTilesAndWritePerSampleOutputs(barcodes);
+        try {
+            basecallsConverter.processTilesAndWritePerSampleOutputs(barcodes);
+        } catch (IOException e) {
+            return 1;
```

Comment:
Do you want to rethrow a PicardException here with some context-specific error information?